src/Compiler/CMakeLists.txt

```diff
@@ -29,10 +24,56 @@ add_custom_target(ExternalUtil DEPENDS ${FILE_GENERATE_DIR}/ExternalUtil.hpp)
 
 get_property(OMLibs GLOBAL PROPERTY ONNX_MLIR_LIBS)
 
-add_onnx_mlir_library(CompilerUtils STATIC
-  CompilerUtils.cpp
+add_onnx_mlir_library(CompilerOptions
+  CompilerOptions.cpp
 
-  EXCLUDE_FROM_OM_LIBS
+  DEPENDS
+  ExternalUtil
+
+  INCLUDE_DIRS PRIVATE
+  ${FILE_GENERATE_DIR}
+  
+  INCLUDE_DIRS PUBLIC
+  ${ONNX_MLIR_SRC_ROOT}/include
+
+  LINK_LIBS PUBLIC
+  MLIRIR
+  )
+
+add_onnx_mlir_library(CompilerPasses
+  CompilerPasses.cpp
+
+  DEPENDS
+  MLIRIR
+
+  INCLUDE_DIRS PRIVATE
+  ${FILE_GENERATE_DIR}
+  
+  INCLUDE_DIRS PUBLIC
+  ${ONNX_MLIR_SRC_ROOT}/include
+
+  LINK_LIBS PUBLIC
+  CompilerOptions
+  OMBundleMemoryPools
```

Comment:
If the various OMLibs are needed for CompilerPasses and marked as public dependencies, you don't need to list them again under CompilerUtils

Reply:
I changed to use OMLibs now. Thanks!